extensions/health/src/controllers/health.controller.ts

```diff
@@ -33,28 +33,57 @@ export function createHealthController(
       responses: {},
       'x-visibility': 'undocumented',
     })
-    health() {
-      return this.healthChecker.getStatus();
+    async health(@inject(RestBindings.Http.RESPONSE) response: Response) {
+      const status = await this.healthChecker.getStatus();
+      return handleStatus(status, response);
     }
 
     @get(options.readyPath, {
       responses: {},
       'x-visibility': 'undocumented',
     })
-    async ready() {
+    async ready(@inject(RestBindings.Http.RESPONSE) response: Response) {
       const status = await this.healthChecker.getReadinessStatus();
-      return status;
+      return handleStatus(status, response);
     }
 
     @get(options.livePath, {
       responses: {},
       'x-visibility': 'undocumented',
     })
-    async live() {
+    async live(@inject(RestBindings.Http.RESPONSE) response: Response) {
       const status = await this.healthChecker.getLivenessStatus();
-      return status;
+      return handleStatus(status, response);
     }
   }
 
   return HealthController;
 }
+
+/**
+ * Create response for the given status
+ * @param status - Health status
+ * @param response - Http response
+ */
+function handleStatus(status: HealthStatus, response: Response) {
```

Comment:
@raymondfeng this might be a bit picky but in general it would make most sense to return a 500 for failing live checks and 503 for failing ready.

503 --> temporary state --> ready fail --> stop routing traffic (temporary)
500 --> something went wrong --> live fail --> restart service

that being said it also assumes that the consumer of the module properly implements the custom checks and for example does not check for dependencies in live checks

Reply:
I don't think k8s treats 500 and 503 differently for health checks. But regardless, I improved to code to return 503/500 respectively.